controllers/datasciencecluster/datasciencecluster_controller.go

```diff
@@ -237,6 +237,16 @@ func (r *DataScienceClusterReconciler) Reconcile(ctx context.Context, req ctrl.R
 			return ctrl.Result{}, err
 		}
 	}
+	// upgrade case to update release version in status
+	if instance.Status.Release.Version.Equals(currentOperatorRelease.Version.Version) {
```

Comment:
If instance has already currentOperatorRelease, why to update it?

Reply:
my bad, i tend to say if not equal....